src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -38,24 +40,35 @@ public override bool ShouldOverrideUrlLoading(AWebView? view, IWebResourceReques
 			var requestUri = request?.Url?.ToString();
 			if (Uri.TryCreate(requestUri, UriKind.RelativeOrAbsolute, out var uri))
 			{
-				if (uri.Host == BlazorWebView.AppHostAddress &&
-					view is not null && 
-					request is not null && 
-					request.IsRedirect && 
-					request.IsForMainFrame)
+				if (AppOriginUri.IsBaseOfPage(uri))
 				{
-					view.LoadUrl(uri.ToString());
-					return true;
+					if (view is not null && 
+						request is not null && 
+						request.IsRedirect && 
+						request.IsForMainFrame)
+					{
+						view.LoadUrl(uri.ToString());
+						return true;
+					}
 				}
-				else if (uri.Host != BlazorWebView.AppHostAddress && _webViewHandler != null)
+				else if (_webViewHandler != null)
 				{
 					var callbackArgs = new ExternalLinkNavigationEventArgs(uri);
 					_webViewHandler.ExternalNavigationStarting?.Invoke(callbackArgs);
 
 					if (callbackArgs.ExternalLinkNavigationPolicy == ExternalLinkNavigationPolicy.OpenInExternalBrowser)
 					{
```

Comment:
The name `OpenInExternalBrowser` really could mean `OpenInExternalApp`. Should we consider renaming this?

Reply:
Good point. What do you think - `OpenInExternalApp`? `OpenExternal`? `OpenExternally`? I'm fine with any. cc @TanayParikh in case he has opinions since he did the earlier version of this.

Reply:
Ideally the same APIs would have as close-to-equivalent-as-we-can behaviors across all platforms. So if we can do it for iOS too, yes.

> Should we even give an option to open external/internal for intent? What would happen if the user requested OpenInWebview for an intent?

My guess is that developers would only want to use `OpenInExternalBrowser` or `CancelNavigation` for intents. But if they explicitly tell us to `InsecureOpenInWebView`, would it be OK to just let them? I know no good will likely come of it, so they probably shouldn't do it, but if they really want to then I'm not sure it's our responsibility to stop them.

Reply:
> Ideally the same APIs would have as close-to-equivalent-as-we-can behaviors across all platforms. So if we can do it for iOS too, yes.

Assuming we're handling iOS/Windows external schemes in the same relative fashion, I think `OpenExternally` or `OpenExternal` makes sense.

> I know no good will likely come of it, so they probably shouldn't do it, but if they really want to then I'm not sure it's our responsibility to stop them.

Fair enough. Can we just make sure if they _do_ do `OpenInWebView` with an intent it just no-ops instead of crashing (not sure what the current code path will do with intents directly. Once we're in a better state with logging, we may also wish to add some logging in place for this to alert developers of a potentially un-intended code path being executed.

Reply:
> Would this be specific to android schemes, or would it apply to iOS custom schemes as well? Or is it handled differently for iOS?

iOS will also open links with non-browser apps if a URL scheme is registered with another app (this was something I tested). So even currently, `OpenInExternalBrowser` doesn't always guarantee that the link will be opened in a browser app.

> What would happen if the user requested OpenInWebview for an intent?

IIRC, assuming the scheme is valid, the webview will just display a "Webpage not available" page (but I can double-check this on Monday). Assuming that's the case, I agree that it's probably fine to allow that page to display without enforcing anything ourselves.